### PR TITLE
fix Gridview dynamic item width is not respected

### LINKED ISSUE #11055: [Android] Gridview dynamic item width is not respected
### Current behavior

The items are not respecting the ItemWidth property that we set in ItemsWrapGrid.

### Expected behavior

The items are respecting the ItemWidth property that we set in ItemsWrapGrid.

### How to reproduce it (as minimally and precisely as possible)

[GridViewColumns.zip](https://github.com/unoplatform/uno/files/10503538/GridViewColumns.zip)

Launch the app on either Android or iOS
Rotate device (There is a behavior that makes the items take up the whole screen)
You will see different results everytime you rotate

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

Uno.UI 4.6.41

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

### IDE version

Version 17.3.4

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #11055

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?
Setting the ItemWidth for the GridViewItem is not respecting the size

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ItemsWrapGrid/ItemsWrapGrid.cs [hunk 1]
```diff
@@ -31,7 +41,7 @@ VirtualizingPanelLayout IVirtualizingPanel.GetLayouter()
 		{
 			if (_layout == null)
 			{
-				_layout = new ItemsWrapGridLayout();
+				_layout = new ItemsWrapGridLayout();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ItemsWrapGrid/ItemsWrapGrid.cs:44]: This is likely going to fail in CI after Youssef's analyzers :)

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/ItemsWrapGrid/ItemsWrapGrid.cs (hunk 1 above)
Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ItemsWrapGrid/ItemsWrapGrid.cs:44]: :D did not see this! Fixing it 🏃‍♂️🏃‍♂️

Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ItemsWrapGrid/ItemsWrapGrid.cs]: Fixed 😅

**Comment by ajpinedam** [comment]
@jeromelaban any idea why the build is failing with that error "fix formatting"? 

image


The block in question has this.

```csharp
#if !__IOS__
	[Ignore]  // Complaining on this line
#endif
```

**Comment by jeromelaban** [comment]
Yes, it's because you need to fix the formatting :) It's @Youssef1313's fault 🤣

I all seriousness, you likely have a space that's not at the right place, or tabs vs. spaces.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_GridView_Items.cs [hunk 2]
```diff
@@ -15,11 +15,77 @@ namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
...
+			await WindowHelper.WaitForLoaded(gridView);
+			await WindowHelper.WaitForIdle();
+
+			Assert.AreEqual(gridView.ActualWidth / 2, gvi2.LayoutSlot.Width, 1);
+		}
+
+#if !__IOS__
+	[Ignore]
+#endif
 		[TestMethod]
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_GridView_Items.cs:89]: The `[Ignore]` should be indented with two tabs, not just one tab

Comment (Youssef1313) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_GridView_Items.cs]: I went ahead and pushed the formatting fix to your branch @ajpinedam

**Comment by jeromelaban** [comment]
@Mergifyio backport release/stable/4.4

**Pull request merged**